nilearn/decomposition/base.py

```diff
@@ -0,0 +1,461 @@
+
+    mask_args : dict, optional
+        If mask is None, these are additional parameters passed to
+        masking.compute_background_mask or masking.compute_epi_mask
+        to fine-tune mask computation. Please see the related documentation
+        for details.
+
+    memory: instance of joblib.Memory or str
+        Used to cache the masking process.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    memory_level: integer, optional
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    n_jobs: integer, optional
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs', -2 'all CPUs but one', and so on.
+
+    in_memory: boolean,
+        Intermediary unmasked data will be
+        stored as a tempory memory map
+
+    verbose: integer, optional
+        Indicate the level of verbosity. By default, nothing is printed.
+
+    Attributes
+    ----------
+    `mask_img_`: Niimg-like object
+        See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+        The mask of the data. If no mask was given at masker creation, contains
+        the automatically computed mask.
+    """
+
+    def __init__(self, n_components=20,
+                 random_state=None,
+                 mask=None, smoothing_fwhm=None,
+                 standardize=True, detrend=True,
+                 low_pass=None, high_pass=None, t_r=None,
+                 target_affine=None, target_shape=None,
+                 mask_strategy='epi', mask_args=None,
+                 memory=Memory(cachedir=None), memory_level=0,
+                 n_jobs=1, in_memory=True,
+                 verbose=0):
+        self.n_components = n_components
+        self.random_state = random_state
+        self.mask = mask
+
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.detrend = detrend
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.mask_strategy = mask_strategy
+        self.mask_args = mask_args
+        self.memory = memory
+        self.memory_level = memory_level
+        self.n_jobs = n_jobs
+        self.in_memory = in_memory
+        self.verbose = verbose
+
+    def fit(self, imgs, y=None, confounds=None):
+        """Base fit for decomposition estimators : compute the embedded masker
+
+        Parameters
+        ----------
+        imgs: list of Niimg-like objects
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Data on which the mask is calculated. If this is a list,
+            the affine is considered the same for all.
+        """
+        if hasattr(imgs, '__iter__') and len(imgs) == 0:
```

Comment:
@AlexandreAbraham has pointed me to _iter_check_niimg in nilean._utils.niimg_conversion that should be useful here.